### PR TITLE
Added convenience functions to artifacts definition registry and moved some artifact validation logic.

### PR DESCRIPTION
**Pull request by joachimmetz**
[Review on Reviewable](https://reviewable.io/reviews/forensicartifacts/artifacts/150)

### CODE REVIEW

File: definitions/linux.yaml [hunk 1]
```diff
@@ -395,6 +395,16 @@ sources:
 labels: [History Files]
 supported_os: [Linux]
 ---
+name: SSHHostPubKeys
```
Comment (destijl) [inline definitions/linux.yaml]: What's going on here, this is already defined? Need to sync maybe?

**Comment by destijl** [comment]
LGTM

**Comment by destijl** [comment]
Give it a better commit message before you merge.

File: definitions/linux.yaml (hunk 1 above)
Comment (joachimmetz) [inline definitions/linux.yaml]: Not sure I'll check tomorrow morning.

Comment (joachimmetz) [inline definitions/linux.yaml]:
```
git fetch upstream && git pull --rebase upstream master
```
fixed the issue

**Pull request merged**